## Statistical analysis

In [1]:
import pandas as pd
import numpy as np
from src.assessments.poo.analiseEstatistica import AnaliseEstatistica, AnaliseEstatisticaExtendida

In [2]:
csv_location = '/app/src/assessments/poo/jobs_in_data.csv'
jobs_data = pd.read_csv(csv_location)
jobs_data.head()

work_year             job_title                    job_category  \
0       2023  Data DevOps Engineer                Data Engineering   
1       2023        Data Architect  Data Architecture and Modeling   
2       2023        Data Architect  Data Architecture and Modeling   
3       2023        Data Scientist       Data Science and Research   
4       2023        Data Scientist       Data Science and Research   

  salary_currency  salary  salary_in_usd employee_residence experience_level  \
0             EUR   88000          95012            Germany        Mid-level   
1             USD  186000         186000      United States           Senior   
2             USD   81800          81800      United States           Senior   
3             USD  212000         212000      United States           Senior   
4             USD   93300          93300      United States           Senior   

  employment_type work_setting company_location company_size  
0       Full-time       Hybrid          Germany            L  
1       Full-time    In-person    United States            M  
2       Full-time    In-person    United States            M  
3       Full-time    In-person    United States            M  
4       Full-time    In-person    United States            M

### Question 1
#### What is the shape of data? How many observations and columns are there?

In [3]:
jobs_data.shape

(9355, 12)

In [4]:
jobs_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9355 entries, 0 to 9354
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           9355 non-null   int64 
 1   job_title           9355 non-null   object
 2   job_category        9355 non-null   object
 3   salary_currency     9355 non-null   object
 4   salary              9355 non-null   int64 
 5   salary_in_usd       9355 non-null   int64 
 6   employee_residence  9355 non-null   object
 7   experience_level    9355 non-null   object
 8   employment_type     9355 non-null   object
 9   work_setting        9355 non-null   object
 10  company_location    9355 non-null   object
 11  company_size        9355 non-null   object
dtypes: int64(3), object(9)
memory usage: 877.2+ KB


### Answer 
#### There are 9355 lines and 12 columns of data.

### Question 2
#### What observations are null? And how many they are?

In [5]:
jobs_data.isnull().values.any()

False

### Answer 
#### There are no null data.

### Question 3
#### What is the average, the median and the standard deviation of salary and salary in usd?

In [13]:
def print_statistical_analysis(num_list: [float], metric_name: str):
  statistics = AnaliseEstatisticaExtendida(num_list)
  print(f'The {metric_name} average is {statistics.media()}')
  print(f'The {metric_name} median is {statistics.mediana()}')
  print(f'The {metric_name} standard deviation is {statistics.desvioPadrao()}\n')

print_statistical_analysis(jobs_data.salary, 'salary')
print_statistical_analysis(jobs_data.salary_in_usd, 'salary in usd')

The salary average is 149927.98129342598
The salary median is 143860
The salary standard deviation is 63605.435572508606

The salary in usd average is 150299.4955638696
The salary in usd median is 143000
The salary in usd standard deviation is 63173.995270803694



### Answer 
#### The salary average is 149927.98129342598
#### The salary median is 143860
#### The salary standard deviation is 63605.435572508606

#### The salary in usd average is 150299.4955638696
#### The salary in usd median is 143000
#### The salary in usd standard deviation is 63173.995270803694

### Question 4
#### What is the biggest and the smallest salary?

In [17]:
def print_metric_range_analysis(num_list: [float], metric_name: str):
  statistics = AnaliseEstatisticaExtendida(num_list)
  print(f'The biggest {metric_name} is {statistics.maximo()}')
  print(f'The smallest {metric_name} is {statistics.minimo()}\n')

print_metric_range_analysis(jobs_data.salary, 'salary')
print_metric_range_analysis(jobs_data.salary_in_usd, 'salary in usd')

The biggest salary is 450000
The smallest salary is 14000

The biggest salary in usd is 450000
The smallest salary in usd is 15000



### Answer 
#### The biggest salary is 450000
#### The smallest salary is 14000

#### The biggest salary in usd is 450000
#### The smallest salary in usd is 15000

### Question 5
#### What is the job title and the job category with more occurrences?

In [27]:
def get_metric_with_more_occurrences(jobs_data, metric: str):
  groupped_data = jobs_data.groupby(metric).count().work_year.sort_values()
  groupped_data.name = 'values'
  return groupped_data.loc[(groupped_data == groupped_data.max())]

job_title_with_more_occurrences = get_metric_with_more_occurrences(jobs_data, 'job_title')
print(job_title_with_more_occurrences)

job_title
Data Engineer    2195
Name: values, dtype: int64


In [26]:
job_category_with_more_occurrences = get_metric_with_more_occurrences(jobs_data, 'job_category')
print(job_category_with_more_occurrences)

job_category
Data Science and Research    3014
Name: values, dtype: int64


### Answer 
#### The job title with more occurrences is Data Engineer
#### and the job category with more occurrences is Data Science and Research